# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/QueueParamTraits.h
# Commit: 4cff94ca9818
# Full Hash: 4cff94ca98189769f21a855520a345c6c4e5c841
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2021-01-15 03:50:53
# Regressor Bug: 1640607
# File Overlap Count: 1
# Description:
#   Bug 1640607 - Send SurfaceDescriptors for GPU blitting for video-to-webgl. r=lsalzman
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D101061
# ==============================================================================

diff -r 044f8b594c71 -r 4cff94ca9818 dom/canvas/QueueParamTraits.h
--- a/dom/canvas/QueueParamTraits.h	Thu Jan 14 01:20:12 2021 +0000
+++ b/dom/canvas/QueueParamTraits.h	Thu Jan 14 01:23:06 2021 +0000
@@ -398,24 +398,25 @@
 
   template <typename U>
   static QueueStatus Write(ProducerView<U>& view, const ParamType& in) {
-    MOZ_ASSERT(!in.image);
-    const bool isSurf = bool(in.surf);
+    MOZ_RELEASE_ASSERT(!in.image);
+    const bool isDataSurf = bool(in.dataSurf);
     if (!view.WriteParam(in.imageTarget) || !view.WriteParam(in.size) ||
         !view.WriteParam(in.srcAlphaType) || !view.WriteParam(in.unpacking) ||
         !view.WriteParam(in.cpuData) || !view.WriteParam(in.pboOffset) ||
-        !view.WriteParam(isSurf)) {
+        !view.WriteParam(in.imageSize) || !view.WriteParam(in.sd) ||
+        !view.WriteParam(isDataSurf)) {
       return view.GetStatus();
     }
-    if (isSurf) {
-      gfx::DataSourceSurface::ScopedMap map(in.surf,
-                                            gfx::DataSourceSurface::READ);
+    if (isDataSurf) {
+      const auto& surf = in.dataSurf;
+      gfx::DataSourceSurface::ScopedMap map(surf, gfx::DataSourceSurface::READ);
       if (!map.IsMapped()) {
         return QueueStatus::kOOMError;
       }
-      const auto& surfSize = in.surf->GetSize();
+      const auto& surfSize = surf->GetSize();
       const auto stride = *MaybeAs<size_t>(map.GetStride());
-      if (!view.WriteParam(surfSize) ||
-          !view.WriteParam(in.surf->GetFormat()) || !view.WriteParam(stride)) {
+      if (!view.WriteParam(surfSize) || !view.WriteParam(surf->GetFormat()) ||
+          !view.WriteParam(stride)) {
         return view.GetStatus();
       }
 
@@ -430,14 +431,15 @@
 
   template <typename U>
   static QueueStatus Read(ConsumerView<U>& view, ParamType* const out) {
-    bool isSurf;
+    bool isDataSurf;
     if (!view.ReadParam(&out->imageTarget) || !view.ReadParam(&out->size) ||
         !view.ReadParam(&out->srcAlphaType) ||
         !view.ReadParam(&out->unpacking) || !view.ReadParam(&out->cpuData) ||
-        !view.ReadParam(&out->pboOffset) || !view.ReadParam(&isSurf)) {
+        !view.ReadParam(&out->pboOffset) || !view.ReadParam(&out->imageSize) ||
+        !view.ReadParam(&out->sd) || !view.ReadParam(&isDataSurf)) {
       return view.GetStatus();
     }
-    if (isSurf) {
+    if (isDataSurf) {
       gfx::IntSize surfSize;
       gfx::SurfaceFormat format;
       size_t stride;
@@ -445,13 +447,14 @@
           !view.ReadParam(&stride)) {
         return view.GetStatus();
       }
-      out->surf = gfx::Factory::CreateDataSourceSurfaceWithStride(
-          surfSize, format, stride, true);
-      if (!out->surf) {
+      auto& surf = out->dataSurf;
+      surf = gfx::Factory::CreateDataSourceSurfaceWithStride(surfSize, format,
+                                                             stride, true);
+      if (!surf) {
         return QueueStatus::kOOMError;
       }
 
-      gfx::DataSourceSurface::ScopedMap map(out->surf,
+      gfx::DataSourceSurface::ScopedMap map(surf,
                                             gfx::DataSourceSurface::WRITE);
       if (!map.IsMapped()) {
         return QueueStatus::kOOMError;